string_filters.hpp

```diff
@@ -124,6 +128,16 @@ inline constexpr lowercase_t lowercase{};
  */
 struct canonicalize_t {
     std::string process(const std::string& str) const {
+#ifdef CPP20_RANGES_ARE_BROKEN
+        /** gcc 11 doesn't like the range solution, ergo this hack */
+        std::vector<std::string> words;
+        std::istringstream f(str);
+        std::string s;
+        while (std::getline(f, s, ' ')) {
+            words.push_back(s | capitalize);
+        }
+        return strings::join(words, "_");
+#else
```

Comment:
This could actually be tightened up further I believe.
```suggestion
        std::vector<std::string> processed_words;
#ifdef CPP20_RANGES_ARE_BROKEN
        /** gcc 11 doesn't like the range solution, ergo this hack */
        std::istringstream f(str);
        std::string s;
        while (std::getline(f, s, ' ')) {
            processed_words.push_back(s | capitalize);
        }
#else
```

Reply:
Then in the else block, remove the definition of processed_words, and the return, and then